pkg/cmd/roachtest/tests/cdc.go

```diff
@@ -139,174 +143,320 @@ func cdcBasicTest(ctx context.Context, t test.Test, c cluster.Cluster, args cdcT
+		if info.status == "succeeded" || info.status == "failed" {
+			close(completionCh)
 		}
-		verifier.statementTime = info.statementTime
-		changefeedLogger.Printf("started changefeed at (%d) %s\n",
-			verifier.statementTime.UnixNano(), verifier.statementTime)
-		t.Status("watching changefeed")
-		return verifier.pollLatency(ctx, db, jobID, time.Second, workloadCompleteCh)
 	})
+	if err != nil {
+		cj.logger.Printf("completion poller error: %s", err)
+	}
+}
 
-	if args.kafkaChaos {
-		m.Go(func(ctx context.Context) error {
-			period, downTime := 2*time.Minute, 20*time.Second
-			return kafka.chaosLoop(ctx, period, downTime, workloadCompleteCh)
-		})
+func newCDCTester(ctx context.Context, t test.Test, c cluster.Cluster) cdcTester {
+	tester := cdcTester{
+		ctx:          ctx,
+		rt:           t,
+		cluster:      c,
+		crdbNodes:    c.Range(1, c.Spec().NodeCount-1),
+		workloadNode: c.Node(c.Spec().NodeCount),
+		doneCh:       make(chan struct{}),
+		sinkCache:    make(map[sinkType]string),
 	}
+	tester.m = c.NewMonitor(ctx, tester.crdbNodes)
 
-	if args.crdbChaos {
-		chaosDuration, err := time.ParseDuration(args.workloadDuration)
-		if err != nil {
-			t.Fatal(err)
-		}
-		ch := Chaos{
-			Timer:   Periodic{Period: 2 * time.Minute, DownTime: 20 * time.Second},
-			Target:  crdbNodes.RandNode,
-			Stopper: time.After(chaosDuration),
-		}
-		m.Go(ch.Runner(c, t, m))
+	changefeedLogger, err := t.L().ChildLogger("changefeed")
+	if err != nil {
+		t.Fatal(err)
 	}
-	m.Wait()
+	tester.logger = changefeedLogger
 
-	verifier.assertValid(t)
-	workloadEnd := timeutil.Now()
-	if args.targetTxnPerSecond > 0.0 {
-		verifyTxnPerSecond(
-			ctx, c, t, crdbNodes.RandNode(), workloadStart, workloadEnd, args.targetTxnPerSecond, 0.05,
-		)
+	c.Put(ctx, t.Cockroach(), "./cockroach")
+	c.Start(ctx, t.L(), option.DefaultStartOpts(), install.MakeClusterSettings(), tester.crdbNodes)
+
+	db := tester.DB()
+	// With a target_duration of 10s, we won't see slow span logs from changefeeds untils we are > 100s
+	// behind, which is well above the 60s targetSteadyLatency we have in some tests.
+	if _, err := db.Exec(
+		`SET CLUSTER SETTING changefeed.slow_span_log_threshold='30s'`,
+	); err != nil {
+		// We don't hard fail here because, not all versions support this setting
+		t.L().Printf("failed to set cluster setting: %s", err)
 	}
+	if _, err := db.Exec("SET CLUSTER SETTING kv.rangefeed.enabled = true"); err != nil {
+		t.L().Printf("failed to set cluster setting: %s", err)
+	}
+
+	return tester
+}
+
+func (t *cdcTester) startGrafana() {
+	// Setup the prometheus instance on the workload node
+	cfg := (&prometheus.Config{}).
+		WithPrometheusNode(t.workloadNode.InstallNodes()[0]).
+		WithCluster(t.crdbNodes.InstallNodes()).
+		WithNodeExporter(t.crdbNodes.InstallNodes()).
+		WithGrafanaDashboard("https://gist.githubusercontent.com/samiskin/75b10ccd4a08280cf1f5d20e916188ae/raw/a5556a5c0c3d82b3449f3140107366995fa16de3/dashboard.json")
```

Comment:
This is just a simple one I threw together right now, will be tweaking it prior to merge and then I'll put it in as a proper file in this repository and link to that, just wanted to get this up for review.

Reply:
ack.